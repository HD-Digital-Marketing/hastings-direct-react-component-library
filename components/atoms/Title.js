import React from 'react';
import '../../scss/atoms/title.scss'

const Title = (props) => {

    const {
        children,
        titleType = 'h1'
    } = props;
    let title = '';

    switch (titleType) {
        case 'h1':
            title = <h1 className="a-title">{children}</h1>
            break;
        case 'h2':
            title = <h2 className="a-title">{children}</h2>
            break;
        case 'h3':
            title = <h3 className="a-title">{children}</h3>
            break;
        case 'h4':
            title = <h4 className="a-title">{children}</h4>
            break;
        case 'h5':
            title = <h5 className="a-title">{children}</h5>
            break;
        case 'h6':
            title = <h6 className="a-title">{children}</h6>
            break;
        default:
            title = <h1 className="a-title">{children}</h1>
    }

    return (
        title
    )

}

export default Title;