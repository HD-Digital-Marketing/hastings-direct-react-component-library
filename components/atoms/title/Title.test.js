import React from 'react';
import renderer from 'react-test-renderer';
import Title from './Title';

describe('Check the title renders', () => {
    test('it should match the snapshot', () => {

        const tree = renderer
            .create(<Title titleType="h3" theme="primary">I am a h3 title</Title>)
            .toJSON();
        expect(tree).toMatchSnapshot();

    });
});