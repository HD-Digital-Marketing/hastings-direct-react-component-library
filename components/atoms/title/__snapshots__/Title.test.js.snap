// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check the title renders it should match the snapshot 1`] = `
<h3
  className="a-title a-title--primary"
>
  I am a h3 title
</h3>
`;
